packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2810,6 +2855,13 @@ class InputDecoration {
   ///  * [suffixIcon], which is the same but on the trailing edge.
   final Widget? prefixIcon;
 
+  /// The alignment for [prefixIcon].
+  ///
+  /// Used to align the [prefixIcon] relative to the position of the text box.
+  ///
+  /// The default value is [VerticalAlignment.center].
+  final VerticalAlignment? prefixIconAlignment;
```

Comment:
Is it possible to use an `Align` widget to position the actual prefix icon, instead of introducing a new parameter?

Reply:
Actually, that was my first approach. But it's not possible. Most of the layout is controlled by performLayout() which I have edited. Just to be sure I've replaced the Center widget with Align...though it may not actually matter.

Reply:
Using `Align` should work? You'll have to specify the `widthFactor` otherwise the icon is going to take up all the horizontal space. e.g.:

```dart
            TextField(
              decoration: InputDecoration(
                prefixIcon: Align(
                  alignment: Alignment.topCenter,
                  widthFactor: 1,
                  child: Icon(Icons.face),
                ),
                labelText: "Phone number / Email / Username",
              ),
            ),

```

Reply:
Again, since the layout occurs in the `performLayout()` method, that actually doesn't help. Also, `performLayout()` calculates the position based on text direction. `prefixIcon` and `suffixIcon` are needed there either way to calculate the position of other elements too.

Reply:
Is this what you're looking for?
![2020-10-23-100643_614x159_scrot](https://user-images.githubusercontent.com/31859944/97033131-b16bdf80-1517-11eb-8d9d-24f4f661b5fa.png)
![2020-10-23-100613_613x183_scrot](https://user-images.githubusercontent.com/31859944/97033087-a2852d00-1517-11eb-8295-92daf28f1850.png)

Not sure I follow the `performLayout` part. Layout always happens in `performLayout`, but you can tell `InputDecoration` to layout an `Align` widget instead of the real icon.

Reply:
@LongCatIsLooong , are you on latest master? Also what will happen when text direction changes?

Reply:
Sorry, I wasn't specific before.
I was talking about [this `performLayout()`](https://github.com/flutter/flutter/blob/cb67513f29dbf7f41818c9eb8cc42d845e243fca/packages/flutter/lib/src/material/input_decorator.dart#L1283)

Reply:
Ohh got it! @LongCatIsLooong , try setting maxLines to something greater than 3 and try aligning it. [`centerLayout()`](https://github.com/flutter/flutter/blob/cb67513f29dbf7f41818c9eb8cc42d845e243fca/packages/flutter/lib/src/material/input_decorator.dart#L1310-L1313) positions it in the center which almost prevents `Align` to position its children because `Align` is made to take up very little space.

Reply:
With `widthFactor: 1`, the `Align` should be as wide as the icon in it, so it should have the exact same layout horizontally.

![2020-10-23-104733_605x168_scrot](https://user-images.githubusercontent.com/31859944/97036831-42918500-151d-11eb-9b7c-84d07e9e701a.png)

Reply:
Sure, but the vertical alignment passed to Align won't work when there are more then 1 lines since `centerLayout()` won't allow it to take up much space.

Reply:
```dart
import 'package:flutter/material.dart';

void main() => runApp(Request());

class Request extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: Container(
            width: 500,
            child: TextField(
              minLines: 1,
              maxLines: 10,
              decoration: InputDecoration(
                prefixIcon: Align(
                  alignment: Alignment.topLeft,
                  widthFactor: 1,
                  child: Icon(Icons.pages),
                ),
              ),
            ),
          ),
        ),
      ),
    );
  }
}
```

This leads to weird behaviour

Reply:
Ah you're right. We need to 2 layout passes for the Align trick to work, the first pass gets the intrinsic height of all components and decides the height of the container, then do a second layout pass to re-layout the components to match the height of the container.

Reply:
It sounds like it could be a common pitfall? I think we should advise against using `Align`  to position the icon in the documentation and point people to the new parameters you added?

Reply:
Yep, you got it! (Sorry I'm bad at explaining). You're right. Adding Align was my first approach as well. Do you think we should add it in the docs for prefixIcon/suffixIcon, both in the source code and in the website?

Thanks for looking into it and sorry I wasn't able to explain.

Reply:
Maybe we can layout the `prefixIcon` and the `suffixIcon` a second time once we've computed the right container height (like we do in `Column` and `Row`)? This way the API stays the same (no new parameters or breaking changes), and widget composition works as expected? /cc @justinmc